gen/nvml/api.go

```diff
@@ -0,0 +1,28 @@
+/**
+# Copyright 2023 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package nvml
+
+// Info represents a set of additional functions that are not associated with the
+// NVML library as pure bindings.
+type Info interface {
+	Lookup(string) error
+}
+
+// Interface represents the interace for the NVML library.
+type Interface interface {
```

Comment:
@klueska as discussed offline, this is the type that would be extended to include our top-level NVML functions, with the `Info` interface capturing the non-NVML functions such as `Lookup` and any other functions that we require (read-only).

Reply:
In general, I like this approach, but I'm not sure of the name `Info` -- at least not paired with the only existing function of `Lookup`. 

A typical usage would then look like:
```
struct mytype {
    nvml nvml.Interface
}
mytype.nvml = nvml.New(WithLibraryPath("..."))
mytype.nvml.Info().Lookup("someSymbol")
```
One of the two alternatives would be better in my view:
```
mytype.nvml.Library().LookupSymbol("someSymbol")
```
or
```
mytype.nvml.Info().LookupLibrarySymbol("someSymbol")
```

Reply:
I think being less generic and using `Library()` over `Info()` is better. We could always define multiple interfaces which group functions if we need to extend this.